Userland/Applications/Maps/MapWidget.cpp

```diff
@@ -444,9 +438,9 @@ void MapWidget::paint_map(GUI::Painter& painter)
     double offset_x = (longitude_to_tile_x(m_center.longitude, m_zoom) - center_tile_x) * TILE_SIZE;
     double offset_y = (latitude_to_tile_y(m_center.latitude, m_zoom) - center_tile_y) * TILE_SIZE;
 
-    // Draw grid around center tile
-    int grid_width = (width() + TILE_SIZE - 1) / TILE_SIZE;
-    int grid_height = (height() + TILE_SIZE - 1) / TILE_SIZE;
+    // Draw grid around center tile; always pad the dimensions with 2 tiles for left/right and top/bottom edges
+    int grid_width = width() / TILE_SIZE + 2;
```

Comment:
This is really `+ 3` (see next lines) because we pad `1 + map + 1 + 1 (offset)`. So if you can change the comment and the next line a bit it is more clear I think.

Reply:
For the rest it looks very good.

Reply:
Ah you're right, those `+ 1` should have been removed. That will probably reintroduce the issue since the iterator is biased for even-numbered widths/heights. I'll take a look at this soon.